connector.go

```diff
@@ -25,6 +25,38 @@ type connector struct {
 	client *http.Client
 }
 
+type StagingCtx struct {
```

Comment:
I don't think it makes sense to define our own struct that implements context.Context.  There's already a mechanism for passing request-scoped data that transits processes and API boundaries.
Define a key type and values and use context.WithValue.